code/modules/admin/create_poll.dm

```diff
@@ -0,0 +1,126 @@
+/client/proc/create_poll()
+	set category = "Special Verbs"
+	set name = "Create Poll"
+
+	if(!check_rights(R_ADMIN|R_DEBUG|R_DEV))
+		return
+	if(!establish_db_connection(dbcon))
+		src << "<span class='danger'>Failed to establish database connection.</span>"
+		return
+	var/polltype = input("Choose poll type.","Poll Type") in list("Single Option","Text Reply","Rating","Multiple Choice")
+	var/choice_amount = 0
+	switch(polltype)
+		if("Single Option")
+			polltype = "OPTION"
+		if("Text Reply")
+			polltype = "TEXT"
+		if("Rating")
+			polltype = "NUMVAL"
+		if("Multiple Choice")
+			polltype = "MULTICHOICE"
+			choice_amount = input("How many choices should be allowed?","Select choice amount") as num
+	var/starttime = SQLtime()
+	var/endtime = input("Set end time for poll as format YYYY-MM-DD HH:MM:SS. All times in server time. HH:MM:SS is optional and 24-hour. Must be later than starting time for obvious reasons.", "Set end time", SQLtime()) as text
+	if(!endtime)
+		return
+	endtime = sanitizeSQL(endtime)
+	var/DBQuery/query_validate_time = dbcon.NewQuery("SELECT STR_TO_DATE('[endtime]','%Y-%c-%d %T')")
+	if(!query_validate_time.Execute())
+		var/err = query_validate_time.ErrorMsg()
+		log_game("SQL ERROR validating endtime. Error : \[[err]\]\n")
```

Comment:
SQL errors are already logged whenever they happen. So just give an admin message and a message to the user instead?